### PR TITLE
DXCDT-308: Add guide on how to achieve 0 downtime client credential rotation

### PR DESCRIPTION
**Pull request by sergiught**
### 📝 Checklist

- [x] All new/changed/fixed functionality is covered by tests (or N/A)
- [x] I have added documentation for all new/changed functionality (or N/A)

### CODE REVIEW

**Comment by sergiught** [comment]
Good point, please check [`f3ca52e` (#592)](https://github.com/auth0/terraform-provider-auth0/pull/592/commits/f3ca52ef9a668dc9e26e32469303106f9a117691)

File: docs/resources/client.md [hunk 1]
```diff
@@ -102,7 +102,7 @@ resource "auth0_client" "my_client" {
...
 - `client_metadata` (Map of String) Metadata associated with the client, in the form of an object with string values (max 255 chars). Maximum of 10 metadata properties allowed. Field names (max 255 chars) are alphanumeric and may only include the following special characters: `:,-+=_*?"/\()<>@ [Tab] [Space]`.
-- `client_secret_rotation_trigger` (Map of String) Custom metadata for the rotation. The contents of this map are arbitrary and are hashed by the provider. When the hash changes, a rotation is triggered. For example, the map could contain the user making the change, the date of the change, and a text reason for the change. For more info: [rotate-client-secret](https://auth0.com/docs/get-started/applications/rotate-client-secret).
+- `client_secret_rotation_trigger` (Map of String, Deprecated) Custom metadata for the rotation. The contents of this map are arbitrary and are hashed by the provider. When the hash changes, a rotation is triggered. For example, the map could contain the user making the change, the date of the change, and a text reason for the change. For more info: [rotate-client-secret](https://auth0.com/docs/get-started/applications/rotate-client-secret).
```
Comment (willvedd) [inline docs/resources/client.md:105]: If deprecating, we should advise how to migrate to the preferred rotation strategy.

File: internal/auth0/client/resource.go [hunk 2]
```diff
@@ -58,6 +61,9 @@ func NewResource() *schema.Resource {
 					"The contents of this map are arbitrary and are hashed by the provider. When the hash changes, a rotation is triggered. " +
 					"For example, the map could contain the user making the change, the date of the change, and a text reason for the change. " +
 					"For more info: [rotate-client-secret](https://auth0.com/docs/get-started/applications/rotate-client-secret).",
+				Deprecated: "Rotating a client's secret through this attribute is deprecated and it will be removed" +
```
Comment (willvedd) [inline internal/auth0/client/resource.go:64]: IMO this is also an appropriate place to mention the secret rotation guide.

File: templates/guides/client_secret_rotation.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+entry and the operation succeeds when using the new secret, the system should discard the old secret and update the
+configuration in a way that the next secret (the one that succeeded) is considered to be the default/current one going
+forward.
+
+## Process (to rotate the credentials):
+
+### Rotating a client secret
+
+1. Generate a new value for the client secret on behalf of the system associated with the client application record.
+This value should have similar entropy to the client secret values generated by the Auth0 service, minimum 48 chars and
```
Comment (willvedd) [inline templates/guides/client_secret_rotation.md:29]:
```suggestion
This value should have similar entropy to the client secret values generated by the Auth0 service, minimum of 48 chars and
```

File: templates/guides/client_secret_rotation.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+valid chars are numbers, letters and `_`, `-`, `+`, `=`, `.` symbols.
```
Comment (willvedd) [inline templates/guides/client_secret_rotation.md:30]: Is there a command or standard process for generating this that we can recommend people?

File: templates/resources/client_credentials.md.tmpl [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+description: |-
+{{ .Description | plainmarkdown | trimspace | prefixlines "  " }}
+---
+
+# {{.Type}}: {{.Name}}
+
+{{ .Description | trimspace }}
+
+-> Check out [our guide](https://registry.terraform.io/providers/auth0/auth0/latest/docs/guides/client_secret_rotation)
+on how to achieve zero downtime client credentials rotation.
```
Comment (willvedd) [inline templates/resources/client_credentials.md.tmpl:12]:
```suggestion
Refer to the [client secret rotation guide](https://registry.terraform.io/providers/auth0/auth0/latest/docs/guides/client_secret_rotation) for instructions on how to rotate client secrets with zero downtime.
```

File: templates/guides/client_secret_rotation.md (hunk 4 above)
Comment (sergiught) [reply templates/guides/client_secret_rotation.md:30]: The random provider could be used https://registry.terraform.io/providers/hashicorp/random/latest/docs.

**Review by willvedd — APPROVED** [review]

**Pull request merged**